lib/components/normal-components/Led.ts

```diff
@@ -15,15 +15,45 @@ export class Led extends NormalComponent<
   get config() {
     return {
       componentName: "Led",
-      schematicSymbolName: this.props.symbolName ?? "led_horz" as BaseSymbolName,
+      schematicSymbolName:
+        this.props.symbolName ?? ("led_horz" as BaseSymbolName),
       zodProps: ledProps,
       sourceFtype: "simple_diode" as Ftype,
     }
   }
 
   initPorts() {
-    this.add(new Port({ name: "pin1", pinNumber: 1, aliases: ["anode", "pos", "left"] }))
-    this.add(new Port({ name: "pin2", pinNumber: 2, aliases: ["cathode", "neg", "right"] }))
+    this.add(
+      new Port({
+        name: "pin1",
+        pinNumber: 1,
+        aliases: ["anode", "pos", "left"],
+      }),
+    )
+    this.add(
+      new Port({
+        name: "pin2",
+        pinNumber: 2,
+        aliases: ["cathode", "neg", "right"],
+      }),
+    )
+  }
+
+  doInitialSchematicComponentRender() {
+    const { db } = this.root!
+    const { _parsedProps: props } = this
+
+    const schematic_component = db.schematic_component.insert({
+      source_component_id: this.source_component_id!,
+      symbol_name: this.props.symbolName ?? ("led_horz" as BaseSymbolName),
```

Comment:
these are sort of supposed to come from the `config`, we're repeating the code a lot here. I think people will copy this pattern for new components and we'll have a lot of repeated code, is it possible to just use the config and have `doInitialSchematicComponentRender` be part of `NormalComponent`?